src/quick-lint-js/parse.h

```diff
@@ -3412,8 +3449,13 @@ class parser {
 #if QLJS_HAVE_SETJMP
   bool have_unimplemented_token_jmp_buf_ = false;
   std::jmp_buf unimplemented_token_jmp_buf_;
+  bool have_parser_depth_limit_exceeded_jmp_buf = false;
+  std::jmp_buf parser_depth_limit_exceeded_jmp_buf_;
 #endif
 
+  std::uint16_t depth_ = 0;
+  const std::uint16_t limit_ = 300;
```

Comment:
Don't allocate `limit_` within the object:
```suggestion
  static constexpr std::uint16_t limit_ = 300;
```